core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.jupiter.api.Test;
+
+import java.time.*;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+public class LocalDateTimeToZonedDateTimeUnitTest {
+
+    @Test
+    void whenConvertLocalDateTimeToZonedDateTimeWithAtZoneMethod_shouldConvertAndMaintainDateTimeValues(){
+        LocalDateTime localDateTime = LocalDateTime.of(2022, 1, 1, 0, 30, 22);
+        ZonedDateTime zonedDateTime = localDateTime.atZone(ZoneId.of("Canada/Atlantic"));
+
+        assertEquals(localDateTime.getYear(), zonedDateTime.getYear());
+        assertEquals(localDateTime.getMonth(), zonedDateTime.getMonth());
+        assertEquals(localDateTime.getDayOfMonth(), zonedDateTime.getDayOfMonth());
+        assertEquals(localDateTime.getHour(), zonedDateTime.getHour());
+        assertEquals(localDateTime.getMinute(), zonedDateTime.getMinute());
+        assertEquals(localDateTime.getSecond(), zonedDateTime.getSecond());
+    }
+
+    @Test
+    void whenConvertLocalDateTimeToZonedDateTimeWithAtZoneAndwithZoneSameInstantMethods_shouldConvert(){
+        LocalDateTime localDateTime = LocalDateTime.of(2022, 1, 1, 0, 30, 22);
+        ZonedDateTime zonedDateTime = localDateTime.atZone(ZoneId.of("Africa/Lagos")).withZoneSameInstant(ZoneId.of("Canada/Atlantic"));
+
+        assertEquals("2022-01-01T00:30:22", localDateTime.toString());
```

Comment:
We probably don't need this assertion and others like it below that involve asserting something on the given conditions, which we already know.

Reply:
@KevinGilmore I was advised to include this by the editor. I reckon that it improves the reader's understanding, so we can just leave it as is.